src/hooks/useOnClickOutside.js

```diff
@@ -1,21 +1,28 @@
 import { useEffect } from 'react'
 
-export default function useOnClickOutside(ref, handler) {
+export default function useOnClickOutside(ref, handler, open) {
+  const clickEvent = 'ontouchstart' in window ? 'touchstart' : 'mousedown'
+
   useEffect(() => {
     const listener = event => {
-      if (!ref.current || ref.current.contains(event.target)) {
+      if (ref.current.contains(event.target)) {
+        // Inside click, return
         return
       }
 
+      // Outside click, Handle event
       handler(event)
     }
 
-    document.addEventListener('mousedown', listener)
-    document.addEventListener('touchstart', listener)
+    // Optimize
```

Comment:
Can we clear up this comment? Your code seems pretty clear to me what the outcome here is.